packages/react/src/autocomplete/root/AutocompleteRoot.test.tsx

```diff
@@ -269,6 +269,69 @@ describe('<Autocomplete.Root />', () => {
         expect(input).to.have.attribute('aria-activedescendant');
       });
     });
+
+    it('retains highlight when clearing the query with autoHighlight enabled', async () => {
+      const { user } = await render(
+        <Autocomplete.Root items={['apple', 'banana', 'cherry']} autoHighlight openOnInputClick>
+          <Autocomplete.Input />
+          <Autocomplete.Portal>
+            <Autocomplete.Positioner>
+              <Autocomplete.Popup>
+                <Autocomplete.List>
+                  {(item: string) => (
+                    <Autocomplete.Item key={item} value={item}>
+                      {item}
+                    </Autocomplete.Item>
+                  )}
+                </Autocomplete.List>
+              </Autocomplete.Popup>
+            </Autocomplete.Positioner>
+          </Autocomplete.Portal>
+        </Autocomplete.Root>,
+      );
+
+      const input = screen.getByRole<HTMLInputElement>('combobox');
+
+      await user.click(input);
+      await user.type(input, 'ban');
+      await waitFor(() => expect(screen.getByRole('listbox')).not.to.equal(null));
+      await waitFor(() => expect(input).to.have.attribute('aria-activedescendant'));
```

Comment:
It should be enough to `waitFor` the first assertion only. 🤔 
Same would apply to all the cases.

```suggestion
      expect(input).to.have.attribute('aria-activedescendant');
```